server/services/store/sqlstore/migrate.go

```diff
@@ -442,6 +466,199 @@ func (s *SQLStore) deleteOldSchemaMigrationTable() error {
 	return nil
 }
 
+// We no longer support boards existing in DMs and private
+// group messages. This function migrates all boards
+// belonging to a DM to the best possible team.
+// The destination is selected as the first team where all members
+// of the DM are a part of. If no such team exists,
+// we use the first team to which DM creator belongs to.
+func (s *SQLStore) migrateTeamLessBoards() error {
+	if !s.isPlugin {
+		return nil
+	}
+
+	setting, err := s.GetSystemSetting(teamLessBoardsMigrationKey)
+	if err != nil {
+		return fmt.Errorf("cannot get teamless boards migration state: %w", err)
+	}
+
+	// If the migration is already completed, do not run it again.
+	if hasAlreadyRun, _ := strconv.ParseBool(setting); hasAlreadyRun {
+		return nil
+	}
+
+	tx, err := s.db.BeginTx(context.Background(), nil)
```

Comment:
Does it make sense to start the transaction after the `getDMBoards`, since if the get fails, nothing to rollback and keep transaction short as possible.

Reply:
Yes, great tip! Done.